### PR TITLE
Logs & errors review

### PR DESCRIPTION
**Pull request by ldez**
### Description

- log & error: remove format if not necessary, add if necessary.
- add constants for k8s annotations.
- fix typos

### CODE REVIEW

File: provider/docker/docker.go [hunk 1]
```diff
@@ -705,7 +705,7 @@ func getLabel(container dockerData, label string) (string, error) {
 			return value, nil
 		}
 	}
-	return "", errors.New("Label not found:" + label)
+	return "", fmt.Errorf("Label not found: %s", label)
```
Comment (timoreimann) [inline provider/docker/docker.go]: Label -> label

File: provider/docker/docker.go [hunk 2]
```diff
@@ -715,7 +715,7 @@ func getLabels(container dockerData, labels []string) (map[string]string, error)
 		foundLabel, err := getLabel(container, label)
 		// Error out only if one of them is defined.
 		if err != nil {
-			globalErr = errors.New("Label not found: " + label)
+			globalErr = fmt.Errorf("Label not found: %s", label)
```
Comment (timoreimann) [inline provider/docker/docker.go]: Label -> label

File: server/configuration.go [hunk 3]
```diff
@@ -88,7 +88,7 @@ func (dep *DefaultEntryPoints) String() string {
 func (dep *DefaultEntryPoints) Set(value string) error {
 	entrypoints := strings.Split(value, ",")
 	if len(entrypoints) == 0 {
-		return errors.New("Bad DefaultEntryPoints format: " + value)
+		return fmt.Errorf("Bad DefaultEntryPoints format: %s", value)
```
Comment (timoreimann) [inline server/configuration.go]: Bad -> bad

File: server/configuration.go [hunk 4]
```diff
@@ -127,7 +127,7 @@ func (ep *EntryPoints) Set(value string) error {
 	regex := regexp.MustCompile("(?:Name:(?P<Name>\\S*))\\s*(?:Address:(?P<Address>\\S*))?\\s*(?:TLS:(?P<TLS>\\S*))?\\s*((?P<TLSACME>TLS))?\\s*(?:CA:(?P<CA>\\S*))?\\s*(?:Redirect.EntryPoint:(?P<RedirectEntryPoint>\\S*))?\\s*(?:Redirect.Regex:(?P<RedirectRegex>\\S*))?\\s*(?:Redirect.Replacement:(?P<RedirectReplacement>\\S*))?\\s*(?:Compress:(?P<Compress>\\S*))?")
 	match := regex.FindAllStringSubmatch(value, -1)
 	if match == nil {
-		return errors.New("Bad EntryPoints format: " + value)
+		return fmt.Errorf("Bad EntryPoints format: %s", value)
```
Comment (timoreimann) [inline server/configuration.go]: Bad -> bad

File: server/configuration.go [hunk 5]
```diff
@@ -314,7 +314,7 @@ func (certs *Certificates) Set(value string) error {
 	for _, certificate := range certificates {
 		files := strings.Split(certificate, ",")
 		if len(files) != 2 {
-			return errors.New("Bad certificates format: " + value)
+			return fmt.Errorf("Bad certificates format: %s", value)
```
Comment (timoreimann) [inline server/configuration.go]: Bad -> bad

File: server/rules.go [hunk 6]
```diff
@@ -167,12 +167,12 @@ func (r *Rules) parseRules(expression string, onRule func(functionName string, f
 		// get function
 		parsedFunctions := strings.FieldsFunc(rule, f)
 		if len(parsedFunctions) == 0 {
-			return errors.New("Error parsing rule: '" + rule + "'")
+			return fmt.Errorf("Error parsing rule: '%s'", rule)
```
Comment (timoreimann) [inline server/rules.go]: Error -> error

File: server/rules.go [hunk 7]
```diff
@@ -167,12 +167,12 @@ func (r *Rules) parseRules(expression string, onRule func(functionName string, f
...
 		}
 		functionName := strings.TrimSpace(parsedFunctions[0])
 		parsedFunction, ok := functions[functionName]
 		if !ok {
-			return errors.New("Error parsing rule: '" + rule + "'. Unknown function: '" + parsedFunctions[0] + "'")
+			return fmt.Errorf("Error parsing rule: '%s'. Unknown function: '%s'", rule, parsedFunctions[0])
```
Comment (timoreimann) [inline server/rules.go]: Error -> error

File: server/rules.go [hunk 8]
```diff
@@ -181,7 +181,7 @@ func (r *Rules) parseRules(expression string, onRule func(functionName string, f
 		// get function
 		parsedArgs := strings.FieldsFunc(strings.Join(parsedFunctions, ":"), fargs)
 		if len(parsedArgs) == 0 {
-			return errors.New("Error parsing args from rule: '" + rule + "'")
+			return fmt.Errorf("Error parsing args from rule: '%s'", rule)
```
Comment (timoreimann) [inline server/rules.go]: Error -> error

File: server/rules_test.go [hunk 9]
```diff
@@ -19,15 +19,14 @@ func TestParseOneRule(t *testing.T) {
 	routeResult, err := rules.Parse(expression)
 
 	if err != nil {
-		t.Fatal("Error while building route for Host:foo.bar")
+		t.Fatal("Error while building route for Host:foo.bar", err)
```
Comment (timoreimann) [inline server/rules_test.go]: Better to use `t.Fatalf`.

File: server/rules_test.go [hunk 10]
```diff
@@ -19,15 +19,14 @@ func TestParseOneRule(t *testing.T) {
...
 	}
 
 	request, err := http.NewRequest("GET", "http://foo.bar", nil)
 	routeMatch := routeResult.Match(request, &mux.RouteMatch{Route: routeResult})
 
 	if routeMatch == false {
-		t.Log(err)
-		t.Fatal("Rule Host:foo.bar don't match")
+		t.Fatal("Rule Host:foo.bar don't match", err)
```
Comment (timoreimann) [inline server/rules_test.go]: Better to use `t.Fatalf`.

File: server/rules_test.go [hunk 11]
```diff
@@ -41,23 +40,21 @@ func TestParseTwoRules(t *testing.T) {
 	routeResult, err := rules.Parse(expression)
 
 	if err != nil {
-		t.Fatal("Error while building route for Host:foo.bar;Path:/FOObar")
+		t.Fatal("Error while building route for Host:foo.bar;Path:/FOObar", err)
```
Comment (timoreimann) [inline server/rules_test.go]: Better to use `t.Fatalf`.

File: server/rules_test.go [hunk 12]
```diff
@@ -41,23 +40,21 @@ func TestParseTwoRules(t *testing.T) {
...
 	}
 
 	request, err := http.NewRequest("GET", "http://foo.bar/foobar", nil)
 	routeMatch := routeResult.Match(request, &mux.RouteMatch{Route: routeResult})
 
 	if routeMatch == true {
-		t.Log(err)
-		t.Fatal("Rule Host:foo.bar;Path:/FOObar don't match")
+		t.Fatal("Rule Host:foo.bar;Path:/FOObar don't match", err)
```
Comment (timoreimann) [inline server/rules_test.go]: Better to use `t.Fatalf`.

File: server/rules_test.go [hunk 13]
```diff
@@ -41,23 +40,21 @@ func TestParseTwoRules(t *testing.T) {
...
 	}
 
 	request, err = http.NewRequest("GET", "http://foo.bar/FOObar", nil)
 	routeMatch = routeResult.Match(request, &mux.RouteMatch{Route: routeResult})
 
 	if routeMatch == false {
-		t.Log(err)
-		t.Fatal("Rule Host:foo.bar;Path:/FOObar don't match")
+		t.Fatal("Rule Host:foo.bar;Path:/FOObar don't match", err)
```
Comment (timoreimann) [inline server/rules_test.go]: Better to use `t.Fatalf`.

File: server/rules_test.go [hunk 14]
```diff
@@ -92,48 +89,48 @@ func TestPriorites(t *testing.T) {
 	rules := &Rules{route: &serverRoute{route: router.NewRoute()}}
 	routeFoo, err := rules.Parse("PathPrefix:/foo")
 	if err != nil {
-		t.Fatal("Error while building route for PathPrefix:/foo")
+		t.Fatal("Error while building route for PathPrefix:/foo", err)
```
Comment (timoreimann) [inline server/rules_test.go]: Better to use `t.Fatalf`.

File: server/rules_test.go [hunk 15]
```diff
@@ -92,48 +89,48 @@ func TestPriorites(t *testing.T) {
...
 	}}, &mux.RouteMatch{}) {
-		t.Fatalf("Error matching route")
+		t.Fatal("Error matching route")
 	}
 
 	multipleRules := &Rules{route: &serverRoute{route: router.NewRoute()}}
 	routeFoobar, err := multipleRules.Parse("PathPrefix:/foobar")
 	if err != nil {
-		t.Fatal("Error while building route for PathPrefix:/foobar")
+		t.Fatal("Error while building route for PathPrefix:/foobar", err)
```
Comment (timoreimann) [inline server/rules_test.go]: Better to use `t.Fatalf`.

File: server/rules_test.go (hunk 9 above)
Comment (ldez) [inline server/rules_test.go]: why?

Comment (timoreimann) [inline server/rules_test.go]: For consistency reasons: this PR already changes a few

`t.Error("something" + err)`

into

`t.Errorf("something: %s", err)`

so it'd only be consequential to stick to the latter format here too.

File: server/configuration.go (hunk 3 above)
Comment (ldez) [inline server/configuration.go]: Why put the first letter in lower case? For me, a sentence begins with an uppercase letter. If it's for consistency, I can change the others too.

Comment (timoreimann) [inline server/configuration.go]: Because of [this](https://github.com/golang/go/wiki/CodeReviewComments#error-strings).

Comment (ldez) [inline server/configuration.go]: Good to known! 👍

**Comment by ldez** [comment]
ping @timoreimann 😉

**Pull request merged**